src/include/duckdb/function/copy_function.hpp

```diff
@@ -107,6 +107,10 @@ typedef idx_t (*copy_desired_batch_size_t)(ClientContext &context, FunctionData
 
 typedef idx_t (*copy_file_size_bytes_t)(GlobalFunctionData &gstate);
 
+typedef bool (*copy_rotate_files_t)(FunctionData &bind_data);
+
+typedef bool (*copy_rotate_next_file_t)(GlobalFunctionData &gstate, FunctionData &bind_data);
```

Comment:
Maybe `copy_file_size_bytes` can be superseded and make use of this callback as well?